src/mgmt/srv_pool.c

```diff
@@ -162,6 +162,8 @@ ds_mgmt_tgt_pool_create_ranks(uuid_t pool_uuid, char *tgt_dev,
 		D_DEBUG(DB_TRACE, "fill ranks %d idx %d "DF_UUID"\n",
 			tc_out_ranks[i], idx, DP_UUID(tc_out_uuids[i]));
 	}
+	D_FREE(tc_out->tc_tgt_uuids.ca_arrays);
+	D_FREE(tc_out->tc_ranks.ca_arrays);
```

Comment:
I think this is incorrect, it is fields in rpc's output struct right? it should be allocated and freed by proc func internally.
Here the test did not report error just because D_FREE set ca_arrays as NULL, so did not trigger issue as free(NULL) will not cause problem.
But the code here looks not correct to me, should remove the 2 D_FREE above.

Reply:
As we discussed this before those arrays are not freed automatically by proc functions now. I need to rework proc code to implement this. For now only decoded by proc structures are freed automatically. Those arrays are encoded only.

Reply:
hmm, there maybe some other issue, I'll do some test and will let you know.

Reply:
These are errors reported by the log-analysis so if they are being freed elsewhere then it's with free(), not D_FREE().

Reply:
I see the problem, actually here you are right, the above two D_FREE() are needed.
it is bcast corpc, at the corpc's initiator the output struct is not processed by proc function, actually it is a result of corpc's co_aggregate callback, here the tc_tgt_uuids/tc_ranks  are allocated by ds_mgmt_tgt_create_aggregator().
Your above code fixed the mem-leak at bcast initiator node, but still with memleak at leaf nodes or middle nodes, for example in ds_mgmt_hdlr_tgt_create() the D_ALLOC_PTR(tmp_tgt_uuid) and D_ALLOC_PTR(rank), in ds_mgmt_tgt_create_aggregator() the D_ALLOC_ARRAY(new_uuids, new_uuids_nr) and D_ALLOC_ARRAY(new_ranks, new_uuids_nr) are all leaked.
I think can free them by providing a co_post_reply() callback. Do you prefer I to submit another patch to fix that memleak, or would you like to add it by yourself?

Reply:
@liuxuezhao, thanks for investigate this. If you know how to free them more correctly please provide the patch. I will appreciate this. If it's small we can just expand this patch with your code.

Reply:
I pushed https://github.com/daos-stack/daos/pull/5037 for that, note that the above two DFREE above still needed.
With PR5037 and your D_FREE above, I verified that all mem mentioned above are freed, on all nodes.